packages/sources/coinmarketcap-test/src/endpoint/crypto.ts

```diff
@@ -1,25 +1,270 @@
-import { CryptoPriceEndpoint } from '@chainlink/external-adapter-framework/adapter'
-import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
 import {
-  buildBatchedRequestBody,
-  constructEntry,
-  CryptoEndpointTypes,
-  inputParameters,
-} from '../crypto-utils'
+  CryptoPriceEndpoint,
+  PriceEndpointInputParameters,
+} from '@chainlink/external-adapter-framework/adapter'
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { makeLogger, SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { InputParameters } from '@chainlink/external-adapter-framework/validation'
+import { config } from '../config'
 import overrides from '../config/overrides.json'
+import presetIds from '../config/presetids.json'
+
+const logger = makeLogger('CryptoCMCEndpoint')
+
+interface CryptoRequestParams {
+  base: string
+  quote: string
+  cid?: string
+  slug?: string
+  resultPath: 'price' | 'volume_24h' | 'market_cap'
+}
+
+const inputParameters = {
+  base: {
+    aliases: ['from', 'coin', 'sym', 'symbol'],
+    description: 'The symbol of symbols of the currency to query',
+    required: true,
+    type: 'string',
+  },
+  quote: {
+    aliases: ['to', 'market', 'convert'],
+    description: 'The symbol of the currency to convert to',
+    required: true,
+    type: 'string',
+  },
+  cid: {
+    description: 'The CMC coin ID (optional to use in place of base)',
+    required: false,
+    type: 'string',
+  },
+  slug: {
+    description: 'The CMC coin ID (optional to use in place of base)',
+    required: false,
+    type: 'string',
+  },
+  resultPath: {
+    description: 'The path to the result within the provider response',
```

Comment:
Nitpick: This isn't really the result within the provider response but don't want to call it the endpoint to avoid confusion. Maybe we can leave it